open-craft/edx-platform
# [BB-6825] Add option to backfill tabs without publish

## Description

This PR adds an option to the `backfill_course_tabs` management command to skip course publish. Currently, running the command triggers course publish for every single course. This could be a very resource-heavy operation in case there are a lot of older courses. The changes introduced here, updates the default tabs in the `CourseOverviewTab` object and exits without publishing the course. It also updates the `get_course_tab_list` method, which is used to fetch the tabs to display in the course page.
Note that, this option cannot be used if you want the tabs list to be updated in the `CourseBlock` object. That still requires a course publish.

## Supporting information

- Sandbox URL: https://maple-nutmeg-dates.opencraft.hosting/ (you'll need django shell access)
- `Private ref`: Jira ticket - [BB-6825](https://tasks.opencraft.com/browse/BB-6825)

## Testing instructions

1. Deploy this branch on your devstack or use the sandbox mentioned above for testing.
2. Import a course to studio from the previous release (i.e. maple).
3. Verify that the "Dates" tab is not visible in the course page in LMS.
4. Run the management command to backfill default course tabs with the option to skip course publish, as shown below:
```
python manage.py cms backfill_course_tabs --no-publish
```
5. Verify that the "Dates" tab is now visible in the course page in LMS and that no course publish was triggered.


## Reviewers
- [ ] @kaustavb12
